src/autogluon/cloud/backend/timeseries_sagemaker_backend.py

```diff
@@ -27,12 +27,15 @@ def _preprocess_data(
         cols = data.columns.to_list()
         # Make sure id and timestamp columns are the first two columns, and target column is in the end
         # This is to ensure in the container we know how to find id and timestamp columns, and whether there are static features being merged
-        timestamp_index = cols.index(timestamp_column)
-        cols.insert(0, cols.pop(timestamp_index))
-        id_index = cols.index(id_column)
-        cols.insert(0, cols.pop(id_index))
-        target_index = cols.index(target)
-        cols.append(cols.pop(target_index))
+        if timestamp_column is not None:
+            timestamp_index = cols.index(timestamp_column)
+            cols.insert(0, cols.pop(timestamp_index))
+        if id_column is not None:
+            id_index = cols.index(id_column)
+            cols.insert(0, cols.pop(id_index))
+        if target is not None:
+            target_index = cols.index(target)
+            cols.append(cols.pop(target_index))
         data = data[cols]
 
         if static_features is not None:
```

Comment:
Making `id_column` an optional argument might break the `pd.merge` in line 45.

I think that keeping the `id_column`, `timestamp_column` and `target` as part of the `TimeSeriesSagemakerBackend` API is fine since this class is not user-facing. In the public API of the `TimeSeriesCloudPredictor` these arguments [are optional](https://github.com/autogluon/autogluon-cloud/blob/master/src/autogluon/cloud/predictor/timeseries_cloud_predictor.py#L53).